### DISCUSSION TITLE
About using in substrate

### DISCUSSION THREAD

**Discussion by Euraxluo** [RingsNetwork/rings#504] [Ideas]
My ultimate goal is to achieve an L1 that can perform ML federated learning and ZKML。

I wanted to see if I could use some of the features in rings-node for P2P  TB-level data transfer， with substrate.

The reason for using substrate is that a lot of subsequent functionality depends on substrate.

**Comment by Euraxluo** [comment]
About this idea, if you want to chat in detail， can add my wechat:euraxluo

**Comment by RyanKung** [comment]
> My ultimate goal is to achieve an L1 that can perform ML federated learning and ZKML。
> 
> I wanted to see if I could use some of the features in rings-node for P2P TB-level data transfer， with substrate.
> 
> The reason for using substrate is that a lot of subsequent functionality depends on substrate.

Hi, the Rings Network can be used for TB-level data transfer, but it uses the WebRTC protocol for its Transport layer, primarily for compatibility with Browser scenarios. 

You mentioned that you are using Substrate as a framework. Could you explain in detail the limitations that libp2p, included in Substrate, has caused for you? 

If your requirement is to transfer TB-level data between browsers, then the Rings Network would be a perfect choice.
